### PR TITLE
Write up basic hammer design

### LINKED ISSUE #20: Tessera Conformance Testing Tool design
Requmts to be fleshed out as part of design. I believe the core idea was to ensure the API and behavior meets the spec. Presumably this would be integrated into CI for each implementation.

### PR DESCRIPTION
**Pull request by mhutchinson**
This provides a mile-high view of the intended design and usage in an
attempt to guide new maintainers.

Fixes #20

### CODE REVIEW

File: hammer/README.md [hunk 1]
```diff
@@ -44,3 +44,40 @@ go run ./hammer \
...
+  - UI: displays the current state of the system and allows some control over the number of workers
+
+The TUI is only intended to be used to interactively explore the capabilities of a log; actual load testing should be done headlessly.
+
+## Deployment
+
+There are 2 main modes that the hammer has been designed to run in:
+  1. Headless, which has 2 main sub-modes:
+    1. Goal-oriented: given both a timeout and a target number of entries to write, process exits successfully if enough entries can be _written_ to the log before the time expires, or otherwise an error code is returned
+    1. Infinite: keeps running until killed in order to perform long-lived performance tests
```
Comment (roger2hk) [inline hammer/README.md:81]: For some reason, it doesn't display properly.

image

**Review by roger2hk — APPROVED** [review]

File: hammer/README.md (hunk 1 above)
Comment (mhutchinson) [reply hammer/README.md]: Fixed, cheers!

**Review by roger2hk — APPROVED** [review]

File: hammer/README.md [hunk 2]
```diff
@@ -44,3 +44,44 @@ go run ./hammer \
   --leaf_write_goal=2500 \
   --show_ui=false
 ```
+
+# Design
```
Comment (AlCutter) [inline hammer/README.md:48]: nit: in some places I've seen `#` be described as a "title level" heading with guidance that there should only be one of these in a given doc - if you subscribe to that philosophy maybe these headings should all be +1 `#`?

Comment (mhutchinson) [reply hammer/README.md:48]: I use markdown a lot and used to subscribe to this philosophy, but then I got bored of excessive indenting and running out of distinguishable indent levels once you get past 3. I can indent everything more if that's the team decision though?

**Pull request merged**